Bump DataTables version & fix header alignment

pxpm opened the pull request:
## WHY

### BEFORE - What was wrong? What was happening before this PR?

We were lagging ALOT in datatables versions. Major improvements and bug fixes had been made in the meanwhile, and we were not benefiting from it. Eg: the fixed header alignment: https://github.com/Laravel-Backpack/CRUD/issues/4853

The reason was the amount of hours I need to spend to track and fix the issues that popped up just by bumping the datatables version in the cdn url. It's done 👍 

### AFTER - What is happening after this PR?

We are on the bleeding egde of datatables, future upgrades will be WAY easier. It looks to me that the table performs a little better. 

### Is it a breaking change?

I don't think so 🤷‍♂️

pxpm commented on src/resources/views/crud/inc/datatables_logic.blade.php:
```diff
@@ -366,18 +371,36 @@ functionsToRunOnDataTablesDrawEvent: [],
       // when datatables-colvis (column visibility) is toggled
       // rebuild the datatable using the datatable-responsive plugin
       $('#crudTable').on( 'column-visibility.dt',   function (event) {
+        console.log('column-visibility.dt');
```
```suggestion
```

pxpm commented on src/resources/views/crud/inc/datatables_logic.blade.php:
```diff
@@ -378,6 +399,7 @@ functionsToRunOnDataTablesDrawEvent: [],
         // when columns are hidden by reponsive plugin,
         // the table should have the has-hidden-columns class
         crud.table.on( 'responsive-resize', function ( e, datatable, columns ) {
+            console.log('responsive-resize', crud.table.responsive.hasHidden());
```
```suggestion
```

tabacitu commented:
@pxpm I can confirm this PR fixes the alignment issue (omfg finally!), and it works for most CRUDS. 

But I noticed it also breaks a few CRUDs in our demo - Invoices CRUD, Pets CRUD:

![CleanShot 2024-12-05 at 13 14 21@2x](https://github.com/user-attachments/assets/aa7fcad9-1264-4c04-989a-38d1fbbccdc8)

tabacitu commented on src/resources/views/crud/list.blade.php:
```diff
@@ -151,9 +151,6 @@ class="{{ backpack_theme_config('classes.table') ?? 'table table-striped table-h
 
 @section('after_styles')
   {{-- DATA TABLES --}}
```
I'm bringing sexy back!!

![sexy-justin-timberlake-gif](https://github.com/user-attachments/assets/cd9d03bf-c066-4241-8aed-6c423840aa56)

YEAH!

```suggestion
  {{-- DATA TABLES --}}
    @basset('https://cdn.datatables.net/2.1.8/css/dataTables.bootstrap5.min.css')
    @basset("https://cdn.datatables.net/responsive/3.0.3/css/responsive.dataTables.min.css")
    @basset('https://cdn.datatables.net/fixedheader/4.0.1/css/fixedHeader.dataTables.min.css')
```

tabacitu commented on src/resources/views/crud/inc/datatables_logic.blade.php:
```diff
@@ -299,16 +304,16 @@ functionsToRunOnDataTablesDrawEvent: [],
       window.crud.updateUrl(location.href);
 
       // move search bar
```
Search bar no longer works for me... it doesn't actually search - no network request, no error, nothing  👀


https://github.com/user-attachments/assets/d8311a1a-fab0-4f89-9ae5-5f1e6df79d01

pxpm commented:
@tabacitu I've removed the `@push()` to the scripts since they are already included in the `after_scripts` section. 

I've just kept the includes on the same file, because not having datatable assets in the list blade, allow us to include the table (in a near future) in other pages that are not related with the `list.blade.php`. 

I've also fixed the search functionality.

tabacitu reviewed (changes requested):
Can confirm it works now! Thanks Pedro.

But there are some differences in looks here, between the old and new. The changes triggered some changes in the footer that makes it looks like there's be no care given to it. Here's what I mean:

![CleanShot 2025-02-07 at 09 45 33](https://github.com/user-attachments/assets/65d73e7a-c24a-486e-87d3-74ca9c7def20)

And here are a few things that stand out to me right away:

![CleanShot 2025-02-07 at 09 46 39@2x](https://github.com/user-attachments/assets/32b85d48-1f93-4b2a-ba26-ba869237fd6f)

Please polish these last things, then merge it.

pxpm commented:
It's done @tabacitu please give it the last look to make sure I didn't forget to polish anything. From my POV it looks ok 👍

pxpm commented:
super seeded by #5688
